### PR TITLE
Move some rendering code to constexpr

### PR DESCRIPTION
**Pull request by rtissera**

### CODE REVIEW

File: src/emu/rendersw.hxx [hunk 1]
```diff
@@ -29,27 +29,27 @@ private:
...
 
 	// internal helpers
-	static inline bool is_opaque(float alpha) { return (alpha >= (_NoDestRead ? 0.5f : 1.0f)); }
-	static inline bool is_transparent(float alpha) { return (alpha < (_NoDestRead ? 0.5f : 0.0001f)); }
-	static inline rgb_t apply_intensity(int intensity, rgb_t color) { return color.scale8(intensity); }
-	static inline float round_nearest(float f) { return floor(f + 0.5f); }
+	static constexpr bool is_opaque(float alpha) { return (alpha >= (_NoDestRead ? 0.5f : 1.0f)); }
+	static constexpr bool is_transparent(float alpha) { return (alpha < (_NoDestRead ? 0.5f : 0.0001f)); }
+	static constexpr rgb_t apply_intensity(int intensity, rgb_t color) { return color.scale8(intensity); }
+	static constexpr float round_nearest(float f) { return floor(f + 0.5f); }
```
Comment (cuavas) [inline src/emu/rendersw.hxx:35]: Neither `floor` nor `std::floor` is `constexpr` as of C++17.

File: src/emu/rendersw.hxx [hunk 2]
```diff
@@ -29,27 +29,27 @@ private:
 	};
 
 	// internal helpers
-	static inline bool is_opaque(float alpha) { return (alpha >= (_NoDestRead ? 0.5f : 1.0f)); }
-	static inline bool is_transparent(float alpha) { return (alpha < (_NoDestRead ? 0.5f : 0.0001f)); }
-	static inline rgb_t apply_intensity(int intensity, rgb_t color) { return color.scale8(intensity); }
-	static inline float round_nearest(float f) { return floor(f + 0.5f); }
+	static constexpr bool is_opaque(float alpha) { return (alpha >= (_NoDestRead ? 0.5f : 1.0f)); }
+	static constexpr bool is_transparent(float alpha) { return (alpha < (_NoDestRead ? 0.5f : 0.0001f)); }
+	static constexpr rgb_t apply_intensity(int intensity, rgb_t color) { return color.scale8(intensity); }
```
Comment (cuavas) [inline src/emu/rendersw.hxx:34]: Will all compilers accept this?  `rgb_t::scale8` modifies a local, which is something MSVC used to reject.

File: src/emu/rendersw.hxx [hunk 3]
```diff
@@ -128,10 +128,10 @@ private:
 	//  palettized 16bpp source
 	//-------------------------------------------------
 
-	static inline u32 get_texel_palette16(const render_texinfo &texture, s32 curu, s32 curv)
+	static constexpr u32 get_texel_palette16(const render_texinfo &texture, s32 curu, s32 curv)
```
Comment (cuavas) [inline src/emu/rendersw.hxx:131]: Is marking this `constexpr` portable given it uses `rgbaint_t` which uses various vector intrinsics?

File: src/emu/rendersw.hxx [hunk 4]
```diff
@@ -196,9 +196,9 @@ private:
 	//  YCbCr source (pixel is returned as Cr-Cb-Y)
 	//-------------------------------------------------
 
-	static inline u32 get_texel_yuy16(const render_texinfo &texture, s32 curu, s32 curv)
+	static constexpr u32 get_texel_yuy16(const render_texinfo &texture, s32 curu, s32 curv)
```
Comment (cuavas) [inline src/emu/rendersw.hxx:199]: This also uses vector intrinsics.

File: src/emu/rendutil.h [hunk 5]
```diff
@@ -58,7 +58,7 @@ ru_imgformat render_detect_image(util::core_file &file);
     round-to-nearest
 -------------------------------------------------*/
 
-static inline float render_round_nearest(float f)
+constexpr float render_round_nearest(float f)
```
Comment (cuavas) [inline src/emu/rendutil.h:61]: This also uses `floor` which is not `constexpr`.

File: src/emu/rendutil.h [hunk 6]
```diff
@@ -117,7 +117,7 @@ static inline int orientation_add(int orientation1, int orientation2)
     a single RGB component
 -------------------------------------------------*/
 
-static inline float apply_brightness_contrast_gamma_fp(float srcval, float brightness, float contrast, float gamma)
+constexpr float apply_brightness_contrast_gamma_fp(float srcval, float brightness, float contrast, float gamma)
```
Comment (cuavas) [inline src/emu/rendutil.h:120]: `pow` is also not `constexpr`

**Review by cuavas — COMMENTED** [review]
I don’t see how this is correct or portable.  It adds `constexpr` to function that:
* use non-`constexpr` standard library functions
* use target-specific vector intrinsics
* access through non-literal types like `render_texinfo`

**Comment by rtissera** [comment]
Sorry for overlooking some issues, I reverted back problematic ones.

**Pull request merged**